language/def_spec.rb

```diff
@@ -523,6 +523,69 @@ def an_instance_eval_class_method;self;end
     DefSpecNested.an_instance_eval_class_method.should == DefSpecNested
     lambda { Object.an_instance_eval_class_method }.should raise_error(NoMethodError)
   end
+  
+  it "creates a class method when the receiver is an anonymous class" do
+    m = Class.new
+    m.instance_eval do
+      def klass_method
+        :test
+      end
+    end
+    
+    m.klass_method.should == :test
+  end
+  
+  it "creates a class method when instance_eval is within class" do
+    m = Class.new do
+      instance_eval do
+        def klass_method
+          :test
+        end          
+      end
+    end
+    
+    m.klass_method.should == :test
+  end
+end
+
+describe "A method definition inside an instance_exec" do
+  it "creates a class method when the receiver is a class" do
+    DefSpecNested.instance_exec(1) do |param|
+      @stuff = param
+      
+      def an_instance_exec_class_method; @stuff; end
+    end
+
+    DefSpecNested.an_instance_exec_class_method.should == 1
+    lambda { Object.an_instance_exec_class_method }.should raise_error(NoMethodError)
```

Comment:
I like this check, because it prevents to just ignore all blocks (including the one to Class.new) and define the method on Object.
Could you replicate it in other cases and in the `alias` new specs?